src/resize.cc

```diff
@@ -694,8 +696,80 @@ class ResizeWorker : public NanAsyncWorker {
       image = gammaDecoded;
     }
 
+    // Apply normalization
+    if (baton->normalize) {
+      VipsInterpretation typeBeforeNormalize = image->Type;
+      if (typeBeforeNormalize == VIPS_INTERPRETATION_RGB) {
+        typeBeforeNormalize = VIPS_INTERPRETATION_sRGB;
+      }
+
+      // normalize the luminance band in LAB space:
+      VipsImage *lab;
+      if (vips_colourspace(image, &lab, VIPS_INTERPRETATION_LAB, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, lab);
+
+      VipsImage *luminance;
+      if (vips_extract_band(lab, &luminance, 0, "n", 1, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, luminance);
+
+      VipsImage *chroma;
+      if (vips_extract_band(lab, &chroma, 1, "n", 2, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, chroma);
+
+      VipsImage *stats;
+      if (vips_stats(luminance, &stats, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, stats);
+      double min = *VIPS_MATRIX(stats, 0, 0);
+      double max = *VIPS_MATRIX(stats, 1, 0);
+      double f = 100.0 / (max - min);
+      double a = -(min * f);
+
+      VipsImage *luminance100;
+      if (vips_linear1(luminance, &luminance100, f, a, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, luminance100);
+
+      VipsImage *normalized;
+      if (vips_bandjoin2(luminance100, chroma, &normalized, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, normalized);
+
+      VipsImage *origFormat;
+      if (vips_colourspace(normalized, &origFormat, typeBeforeNormalize, NULL)) {
+        return Error();
+      }
+      vips_object_local(hook, origFormat);
+
+      if (HasAlpha(image)) {
+        VipsImage *alpha;
+        if (vips_extract_band(image, &alpha, image->Bands - 1, "n", 1, NULL)) {
+          return Error();
+        }
+        vips_object_local(hook, alpha);
+
+        VipsImage *normalizedAlpha;
+        if (vips_bandjoin2(origFormat, alpha, &normalizedAlpha, NULL)) {
+          return Error();
+        }
+        vips_object_local(hook, normalizedAlpha);
+        image = normalizedAlpha;
+      } else {
+        image = origFormat;
+      }
+    }
+
     // Convert image to sRGB, if not already
-    if (image->Type != VIPS_INTERPRETATION_sRGB) {
+    if (image->Type != VIPS_INTERPRETATION_sRGB && image->Type != VIPS_INTERPRETATION_B_W) {
```

Comment:
I opened jcupitt/libvips#267 to find out wether vips_webpsave should support saving b_w aimages at all.

btw, The line in question breaks more stuff than webp: embedded profiles also cause problems further down.